builder/dockerignore.go

```diff
@@ -26,8 +26,8 @@ type DockerIgnoreContext struct {
 //
 // TODO: Don't require a ModifiableContext (use Context instead) and don't remove
 // files, instead handle a list of files to be excluded from the context.
-func (c DockerIgnoreContext) Process(filesToRemove []string) error {
-	f, err := c.Open(".dockerignore")
+func (c DockerIgnoreContext) Process(ignoreFile string, filesToRemove []string) error {
+	f, err := c.Open(ignoreFile)
 	// Note that a missing .dockerignore file isn't treated as an error
```

Comment:
This would mean making the server side know what the default is (i.e. `.dockerignore`), I think it make sense, even just for retro-compatibility API-wise 👼